hack/build-image-docker.sh

```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/env bash
+
+# Copyright 2025 The KCP Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Build container images for KCP using Docker
+#
+# This script builds container images using Docker (with or without buildx).
+#
+# Usage examples:
+#   # Build locally with default settings (uses current git commit hash)
+#   ./hack/build-image-docker.sh
+#   
+#   # Build locally with custom repository name
+#   REPOSITORY=my-registry/kcp ./hack/build-image-docker.sh
+#   
+#   # Build locally without pushing (dry run)
+#   DRY_RUN=1 ./hack/build-image-docker.sh
+#   
+#   # Build for specific architectures only
+#   ARCHITECTURES="amd64" ./hack/build-image-docker.sh
+#
+# Environment variables:
+#   REPOSITORY - Override default repository (default: ghcr.io/kcp-dev/kcp)
+#   ARCHITECTURES - Space-separated list of architectures (default: "amd64 arm64")
+#   DRY_RUN - Set to any value to build locally without pushing
+#   KCP_GHCR_USERNAME/KCP_GHCR_PASSWORD - Registry credentials for pushing
+#
+# Build tool support:
+#   - docker + buildx: Multi-arch support with intelligent platform handling  
+#   - docker only: Single architecture fallback
+
+set -euo pipefail
+
+# make git available
+if ! [ -x "$(command -v git)" ]; then
+  echo "Installing git ..."
+  yum install -y git
+fi
+
+# Check if docker is available
+if ! [ -x "$(command -v docker)" ]; then
+  echo "Error: Docker is not available"
+  exit 1
+fi
+
+echo "Using docker for container builds"
+# Check if buildx is available for multi-arch builds
+if docker buildx version >/dev/null 2>&1; then
+  echo "Docker buildx is available for multi-arch builds"
+  DOCKER_BUILDX=true
+else
+  echo "Docker buildx not available, falling back to single-arch builds"
+  DOCKER_BUILDX=false
+fi
+
+
+repository=${REPOSITORY:-ghcr.io/kcp-dev/kcp}
```

Comment:
Does it make sense to have this default here? We aren't going to use this in CI after all, and I would like to avoid someone with maintainer permissions to run this by mistake and push images.